src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs

```diff
@@ -0,0 +1,309 @@
+        /// caused this activity.   This is only intended to be used at 'boundary' 
+        /// scenarios where an activity from another process loggically started 
+        /// this activity. The Parent ID shows up the Tags (as well as the ParentID 
+        /// property), and can be used to reconstruct the causal tree.  
+        /// Returns 'this' for convinient chaining.
+        /// </summary>
+        public Activity SetParentId(string parentId)
+        {
+            ParentId = parentId;
+            return this;
+        }
+
+        /// <summary>
+        /// Update the Activity to set start time
+        /// </summary>
+        /// <param name="startTimeUtc">Activity start time in UTC (Greenwitch Mean Time)</param>
+        /// <returns>'this' for convinient chaining</returns>
+        public Activity SetStartTime(DateTime startTimeUtc)
+        {
+            StartTimeUtc = startTimeUtc;
+            return this;
+        }
+        /// <summary>
+        /// Update the Activity to set <see cref="Duration"/>
+        /// as a difference between <see cref="StartTimeUtc"/>
+        /// and given stop timestamp
+        /// </summary>
+        /// <param name="endTimeUtc">Activity stop time in UTC (Greenwitch Mean Time)</param>
+        /// <returns>'this' for convinient chaining</returns>
+        public Activity SetEndTime(DateTime endTimeUtc)
+        {
+            Duration = endTimeUtc - StartTimeUtc;
+            return this;
+        }
+
+        /// <summary>
+        /// If the Activity has ended (Stop was called) then this is the delta
+        /// between start and end.   If the activity is not ended then this is 
+        /// TimeSpan.Zero.  
+        /// </summary>
+        public TimeSpan Duration { get; private set; }
+
+        /// <summary>
+        /// Starts activity: sets <see cref="Parent"/> to hold <see cref="Current"/> and sets Current to this Activity.
+        /// If <see cref="StartTimeUtc"/> was not set previously, sets it to DateTime.UtcNow.
+        /// Use DiagnosticSource.Start to start activity and write start event.
+        /// </summary>
+        /// <returns>Started activity for convinient chaining</returns>
+        /// <seealso cref="DiagnosticSource.StartActivity(Activity, object)"/>
+        /// <seealso cref="SetStartTime(DateTime)"/>
+        public Activity Start()
+        {
+            if (Id != null)
+                throw new InvalidOperationException("Trying to start an Activity that was already started");
+
+            if (ParentId == null)
+            {
+                var parent = Current;
+                if (parent != null)
+                {
+                    ParentId = parent.Id;
+                    Parent = parent;
+                }
+            }
+
+            if (StartTimeUtc == default(DateTime))
+                StartTimeUtc = DateTime.UtcNow;
+
+            Id = GenerateId();
+
+            Current = this;
```

Comment:
I'm not sure if this is a good idea. Setting static properties in instance methods is bad practice and unexpected. Would it be better to have this in DiagnosticSource.StartActivity?!